src/Package/Impl/Help/HelpWindowPane.cs

```diff
@@ -28,5 +37,64 @@ internal class HelpWindowPane : VisualComponentToolWindow<IHelpVisualComponent>
             ToolBarCommandTarget = new CommandTargetToOleShim(null, Component.Controller);
             base.OnCreate();
         }
+
+        public override bool SearchEnabled => true;
+
+        public override void ProvideSearchSettings(IVsUIDataSource pSearchSettings) {
+            var settings = (SearchSettingsDataSource)pSearchSettings;
+            settings.SearchWatermark = Resources.HelpSearchWatermark;
+            settings.SearchTooltip = Resources.HelpSearchTooltip;
+            settings.SearchStartType = VSSEARCHSTARTTYPE.SST_ONDEMAND;
+            base.ProvideSearchSettings(pSearchSettings);
+        }
+
+        public override IVsSearchTask CreateSearch(uint dwCookie, IVsSearchQuery pSearchQuery, IVsSearchCallback pSearchCallback) {
+            return new HelpSearchTask(dwCookie, pSearchQuery, pSearchCallback);
+        }
+
+        private sealed class HelpSearchTask : VsSearchTask {
+            //private static IEnumerable<Lazy<IRHelpSearchTermProvider>> _termProviders;
+            //private readonly List<string> _terms = new List<string>();
+            private readonly IVsSearchCallback _callback;
+            private IRInteractiveWorkflowProvider _workflowProvider;
+
+            public HelpSearchTask(uint dwCookie, IVsSearchQuery pSearchQuery, IVsSearchCallback pSearchCallback)
+                : base(dwCookie, pSearchQuery, pSearchCallback) {
+                _callback = pSearchCallback;
+                _workflowProvider = VsAppShell.Current.ExportProvider.GetExportedValue<IRInteractiveWorkflowProvider>();
+            }
+
+            protected override async void OnStartSearch() {
+                base.OnStartSearch();
+                _callback.ReportComplete(this, 1);
+
+                var searchString = SearchQuery.SearchString;
+                if (!string.IsNullOrWhiteSpace(searchString)) {
+                    var session = _workflowProvider.GetOrCreate().RSession;
+                    using (var inter = await session.BeginInteractionAsync()) {
+                        await inter.RespondAsync("?" + searchString + Environment.NewLine);
```

Comment:
The tricky part with `?` is that it doesn't cause the help to open when it is evaluated. Instead, it returns an object (actually just a string URL with a custom class) that causes help to open when it is printed. So if you just evaluate `?x`, you get that string back, and nothing opens. `show_help` wraps all the stuff that's needed to actually make it open.

On the other hand, if you want to test whether search is successful or not, this means that you can eval `?x` first and check the result. If it didn't find anything, it'll be a 0-length vector. On the other hand, `??` never fails.